src/components/cache-surah-button.tsx

```
@@ -0,0 +1,189 @@
+        return;
+      }
+      // Clear cache functionality would go here
+      return;
+    }
+
+    try {
+      setProgress(0);
+      console.log(`[Cache] Starting to cache surah ${surahId}...`);
+      
+      // Find the URL for this surah from the playlist
+      const surahIndex = parseInt(surahId) - 1; // surahId is 1-indexed
```

ESLint

[error] Prefer `Number.parseInt` over `parseInt`.

(unicorn/prefer-number-properties)